--- a/casper/filesystem.manifest-remove
+++ b/casper/filesystem.manifest-remove
@@ -45,8 +45,6 @@ libfcitx-gclient0:amd64
 libfcitx-qt0:amd64
 libfcitx-qt5-1:amd64
 libfcitx-utils0:amd64
-libgeoclue0:amd64
-libjavascriptcoregtk-4.0-18:amd64
 liblua5.2-0:amd64
 liblvm2app2.2:amd64
 liblvm2cmd2.02:amd64
@@ -55,10 +53,7 @@ libpresage-data
 libpresage1v5:amd64
 libreadline5:amd64
 libtinyxml2.6.2v5:amd64
-libwebkit2gtk-4.0-37:amd64
-libwebkit2gtk-4.0-37-gtk2:amd64
 linux-signed-generic
-linux-signed-image-4.4.0-18-generic
 linux-signed-image-generic
 localechooser-data
 lupin-casper
@@ -86,5 +81,3 @@ ubiquity-slideshow-kubuntu
 ubiquity-ubuntu-artwork
 user-setup
 xfsprogs
-zenity
-zenity-common
