--- a/scripts/casper	2014-03-13 20:35:34.293226677 +0100
+++ b/scripts/casper	2014-03-13 20:28:39.978460056 +0100
@@ -43,6 +43,8 @@
                 export PERSISTENT_PATH="${x#persistent-path=}" ;;
             union=*)
                 export UNIONFS="${x#union=}";;
+            unionstack=*)
+                export UNIONSTACK="${x#unionstack=}";;
             ip=*)
                 STATICIP=${x#ip=}
                 if [ "${STATICIP}" = "" ]; then
@@ -490,6 +492,13 @@
                 continue
             fi
 
+            if [ -n "$UNIONSTACK" ]; then
+                if [ "$(echo $UNIONSTACK | sed "s/$imagename//";)" == "$UNIONSTACK" ]; then
+                    log_warning_msg "Filesystem image $imagename not in UNIONSTACK. Ignored."
+                    continue
+                fi
+            fi
+
             if [ -d "${image}" ]; then
                 # it is a plain directory: do nothing
                 rofsstring="${image}=${roopt}:${rofsstring}"
@@ -519,26 +528,18 @@
             mount -t ${UNIONFS} -o noatime,dirs=/cow/upper=rw:$rofsstring ${UNIONFS} "$rootmnt" || panic "${UNIONFS} mount failed"
             ;;
         overlay|overlayfs)
-            # Mount the layers pairwise from the bottom onto rootmnt,
-            # for the second and later layers rootmnt forms the lower layer.
             mounts=""
-            for mount in /cow $rofslist
+            for mount in $rofslist
             do
-                mounts="$mount $mounts"
-            done
-            lower=""
-            for mount in $mounts
-            do
-                if [ "$lower" = "" ]; then
-                    lower="$mount"
-                    continue
-                fi
-                mount -t ${UNIONFS} -o "upperdir=$mount/upper,lowerdir=$lower,workdir=$mount/work" \
-                    "$mount" "$rootmnt" || \
-                  mount -t ${UNIONFS} -o "upperdir=$mount/upper,lowerdir=$lower" \
-                      "$mount" "$rootmnt"
-                lower="$rootmnt"
+                mounts="${mount}:${mounts}"
             done
+            mounts="${mounts%:}"
+            overlay_mount_dir=/overlay
+            mkdir -p "${overlay_mount_dir}"
+            mount -t ${UNIONFS} -o "lowerdir=${mounts}" root "${overlay_mount_dir}"
+            # Stacking multiple overlays leads to "No such device" errors upon access
+            mount -t ${UNIONFS} -o "lowerdir=${mounts}",upperdir=/cow/upper,workdir=/cow/work" cowroot "${rootmnt}"
+            lower="${overlay_mount_dir}"
             ;;
     esac
 
@@ -579,15 +580,17 @@
     fi
 
     # move the first mount; no head in busybox-initramfs
-    for d in $(mount -t squashfs | cut -d\  -f 3); do
-        mkdir -p "${rootmnt}/rofs"
-        if [ "${UNIONFS}" = unionfs-fuse ]; then
-            mount -o bind "${d}" "${rootmnt}/rofs"
-        else
-            mount -o move "${d}" "${rootmnt}/rofs"
-        fi
-        break
-    done
+    #for d in $(mount -t squashfs | cut -d\  -f 3); do
+    #    mkdir -p "${rootmnt}/rofs"
+    #    if [ "${UNIONFS}" = unionfs-fuse ]; then
+    #        mount -o bind "${d}" "${rootmnt}/rofs"
+    #    else
+    #        mount -o move "${d}" "${rootmnt}/rofs"
+    #    fi
+    #    break
+    #done
+    mkdir -p "${rootmnt}/rofs"
+    mount -o move "$lower" "${rootmnt}/rofs"
 }
 
 check_dev ()
